lib/account/account-manager.js

```diff
@@ -43,60 +87,67 @@ class AccountManager {
-     * @return {Promise.<AccountService~accountFile>}
+     * @return {Promise.<Object>} the accountId is sent back to the user when account file is made
      */
-    completeAsync(accountInfo) {
-        return this.accountService.getDirectoryAsync(accountInfo.accountId, this.config.account).then((directory) => {
+    function signupCompleteAsync(accountInfo) {
+        return accountService.getRegistrationFileAsync(accountInfo.regId).then((accountFile) => {
+            checkMfa(accountFile.mfaKey, accountInfo.previousMfa, accountInfo.currentMfa);
 
-            return this.accountService.getAsync(directory).then((accountFile) => {
-                var options;
-
-                if (! this.hotp.verifyToken(accountFile.mfaKey, accountInfo.previousMfa, 1)) {
-                    throw new Error("Previous MFA Token did not validate");
-                }
-
-                if (! this.hotp.verifyToken(accountFile.mfaKey, accountInfo.currentMfa)) {
-                    throw new Error("Current MFA Token did not validate");
-                }
+            return random.passwordAsync(24);
+        }).then((accountId) => {
+            return setupAccountAsync(accountId, accountInfo);
+        });
+    }
 
-                options = {
-                    expires: this.otDate.now().plus(this.config.account.completeLifetime)
-                };
 
-                return this.accountService.completeAsync(directory, accountInfo, options);
-            });
+    /**
+     * Confirms the account holder wants to confirm the account and get the information
+     * to complete sign up.
+     *
+     * @param {string} regId
+     * @return {Promise.<accountManger~confirm>}
+     */
+    function signupConfirmAsync(regId) {
+        return accountService.getRegistrationFileAsync(regId).then((accountFile) => {
+            return {
+                passwordSalt: accountFile.passwordSalt,
+                mfaKey: accountFile.mfaKey
+            };
         });
     }
 
 
     /**
-     * Initiates the account creation and puts a file on the server
-     * with a hashed accountId.
+     * Initiates the account creation by creating
+     * a registration file with initial data for an account
+     * holder to confirm and complete the account signup with.
      *
      * TODO: Validate email before doing anything else
      *
      * @param {Object} accountInfo
-     * @return {Promise.<accountService~initiateResult>}
+     * @return {Promise.<Object>} The registration ID is returned to the account initiater
      */
-    initiateAsync(accountInfo) {
-        return this.promise.props({
-            accountId: this.random.password(24),
-            mfaKey: this.hotp.generateSecretAsync(),
-            salt: this.random.password(128)
+    function signupInitiationAsync(accountInfo) {
+        return promise.props({
+            mfaKey: hotp.generateSecretAsync(),
+            regId: random.passwordAsync(24),
+            passwordSalt: random.passwordAsync(128)
```

Comment:
Rename `random.passwordAsync` since you are not generating passwords.  You're generating identifiers or random encoded strings.